### ISSUE TITLE
Prometheus crashes when using spec.alerting.alertmanagers.tlsConfig.ca field to specify CA certificate for alertmanager

### ISSUE THREAD

**Issue by vyzigold** [prometheus-operator/prometheus-operator#6760]
### Is there an existing issue for this?

- [X] I have searched the existing issues

## Description
I tried to configure TLS for alertmanager web endpoint. Using curl, I can see the TLS on Alertmanager is working correctly (if I execute: `kubectl exec curl-5c56f88c47-nvkhb -- curl https://sample-monitoring-stack-alertmanager.operators.svc:9093 --cacert /certs/am/ca` I get the expected alertmanager html in the output). I tried to configure Prometheus to access the alertmanager with TLS with using the correct CA certificate. This resulted in Prometheus crashing. I used a debug pod to look inside the prometheus container. According to the generated prometheus config, it expects the CA certificate to be inside /etc/prometheus/certs, but the folder is empty. According to the Prometheus pod spec, it's supposed to mount the tls-assets secret into that folder, but the tls-assets secret is empty. I think there is some bug related to generating that secret from certificates specified in spec.alerting.alertmanagers.tlsConfig.

All the related commands and outputs are below.

Note: I also tried to use the same Alertmanager and Prometheus CRs, but without the TLS configuration and create a PrometheusRule. Everything worked correctly

## Steps to Reproduce

1. Apply certificate secrets
2. Apply Prometheus and Alertmanager CRs
3. Create a Service for Alertmanager
4. Wait for the Prometheus container to start crashing

## Expected Result
Using spec.alerting.alertmanagers.tlsConfig.ca property in Prometheus CR results in a working Prometheus, which can access Alertmanager with TLS.

## Actual Result
Prometheus container crashes.

### Prometheus Operator Version

```shell
Running locally from source from main branch:

git log | head
commit 79c225b4aa97566504f1a5dc72ea1862f9dbe900
Merge: 91df2211c 21e69ffe5
Author: Jayapriya Pai <slashpai9@gmail.com>
Date:   Wed Jun 26 20:19:59 2024 +0530

    Merge pull request #6709 from slashpai/cut-0.75.0
    
    chore: cut 0.75.0
```
```


### Kubernetes Version

```yaml
kubectl version -o yaml
clientVersion:
  buildDate: "2024-06-10T15:10:01Z"
  compiler: gc
  gitCommit: 44edfb50c55756395abc844fc4f7edb19dfefa50
  gitTreeState: clean
  gitVersion: v1.28.2
  goVersion: go1.20.12 X:strictfipsruntime
  major: "1"
  minor: "28"
  platform: linux/amd64
kustomizeVersion: v5.0.4-0.20230601165947-6ce0bf390ce3
serverVersion:
  buildDate: "2024-05-13T22:00:36Z"
  compiler: gc
  gitCommit: 7c48c2bd72b9bf5c44d21d7338cc7bea77d0ad2a
  gitTreeState: clean
  gitVersion: v1.30.0
  goVersion: go1.22.2
  major: "1"
  minor: "30"
  platform: linux/amd64
```

### Kubernetes Cluster Type

kind

### How did you deploy Prometheus-Operator?

Other (please comment)

### Manifests

```yaml
apiVersion: monitoring.coreos.com/v1
kind: Prometheus
metadata:
  creationTimestamp: "2024-07-18T09:19:28Z"
  generation: 4
  name: sample-monitoring-stack
  namespace: operators
  resourceVersion: "25621"
  uid: 9ff84367-f16c-4110-8055-8139358120ce
spec:
  alerting:
    alertmanagers:
    - apiVersion: v2
      name: sample-monitoring-stack-alertmanager
      namespace: operators
      port: web
      scheme: https
      tlsConfig:
        ca:
          secret:
            key: ca.crt
            name: cert-alertmanager-svc
        serverName: sample-monitoring-stack-alertmanager
  arbitraryFSAccessThroughSMs: {}
  evaluationInterval: 30s
  logLevel: debug
  podMetadata:
    labels:
      app.kubernetes.io/component: prometheus
      app.kubernetes.io/part-of: sample-monitoring-stack
  podMonitorSelector:
    matchLabels:
      app: demo
  portName: web
  probeSelector:
    matchLabels:
      app: demo
  replicas: 1
  retention: 1d
  ruleSelector:
    matchLabels:
      app: demo
  rules:
    alert: {}
  scrapeConfigSelector:
    matchLabels:
      app: demo
  scrapeInterval: 30s
  serviceMonitorSelector:
    matchLabels:
      app: demo
status:
  availableReplicas: 0
  conditions:
  - lastTransitionTime: "2024-07-18T13:42:05Z"
    message: 'shard 0: pod prometheus-sample-monitoring-stack-0: containers with unready
      status: [prometheus]'
    observedGeneration: 4
    reason: NoPodReady
    status: "False"
    type: Available
  - lastTransitionTime: "2024-07-18T13:42:05Z"
    message: ""
    observedGeneration: 4
    reason: ""
    status: "True"
    type: Reconciled
  paused: false
  replicas: 1
  selector: app.kubernetes.io/instance=sample-monitoring-stack,app.kubernetes.io/managed-by=prometheus-operator,app.kubernetes.io/name=prometheus,operator.prometheus.io/name=sample-monitoring-stack,prometheus=sample-monitoring-stack
  shardStatuses:
  - availableReplicas: 0
    replicas: 1
    shardID: "0"
    unavailableReplicas: 1
    updatedReplicas: 1
  shards: 1
  unavailableReplicas: 1
  updatedReplicas: 1


apiVersion: monitoring.coreos.com/v1
kind: Alertmanager
metadata:
  creationTimestamp: "2024-07-18T08:50:17Z"
  generation: 3
  name: sample-monitoring-stack
  namespace: operators
  resourceVersion: "21903"
  uid: aedf2839-7ea6-4089-ad7f-40ba0b6b4a30
spec:
  portName: web
  replicas: 1
  retention: 120h
  web:
    tlsConfig:
      cert:
        secret:
          key: tls.crt
          name: cert-alertmanager-svc
      keySecret:
        key: tls.key
        name: cert-alertmanager-svc
status:
  availableReplicas: 1
  conditions:
  - lastTransitionTime: "2024-07-18T12:57:49Z"
    message: ""
    observedGeneration: 3
    reason: ""
    status: "True"
    type: Available
  - lastTransitionTime: "2024-07-18T08:50:17Z"
    message: ""
    observedGeneration: 3
    reason: ""
    status: "True"
    type: Reconciled
  paused: false
  replicas: 1
  unavailableReplicas: 0
  updatedReplicas: 1


# The certificate at the bottom is created for the sample-monitoring-stack-alertmanager.operators... URLs. That's why I included a service with the certificate here.

apiVersion: v1
kind: Service
metadata:
  name: sample-monitoring-stack-alertmanager
  namespace: operators
spec:
  clusterIP: None
  clusterIPs:
  - None
  internalTrafficPolicy: Cluster
  ipFamilies:
  - IPv4
  ipFamilyPolicy: SingleStack
  ports:
  - name: web
    port: 9093
    protocol: TCP
    targetPort: web
  - name: tcp-mesh
    port: 9094
    protocol: TCP
    targetPort: 9094
  - name: udp-mesh
    port: 9094
    protocol: UDP
    targetPort: 9094
  selector:
    app.kubernetes.io/name: alertmanager
  sessionAffinity: None
  type: ClusterIP


apiVersion: v1
data:
  ca.crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUVBRENDQW1pZ0F3SUJBZ0lRR1MyUjRwTGtZTGlMSCthbTNjT1VTVEFOQmdrcWhraUc5dzBCQVFzRkFEQWEKTVJnd0ZnWURWUVFERXc5eWIyOTBZMkV0YVc1MFpYSnVZV3d3SGhjTk1qUXdOVEl4TURZeU5UTTRXaGNOTXpRdwpOVEU1TURZeU5UTTRXakFhTVJnd0ZnWURWUVFERXc5eWIyOTBZMkV0YVc1MFpYSnVZV3d3Z2dHaU1BMEdDU3FHClNJYjNEUUVCQVFVQUE0SUJqd0F3Z2dHS0FvSUJnUUN0MXlyMmhqNnh4VHl4WUxKOTRBMGJFOEo5MGYwTzlFeDMKUHlZTitxT1prUVlOZEtBWHlVa0J5OXJ4Z2hUa09rb3hDc1d0c1R2RDl1bTIvUWMyRHNtc2V3eXBpVGpXSGRwagoybi83TVZmNWFiTzV2OTR4b2l5VFM5NCtNd29qUzU0c280VThXaXVpSFFkRm01UWVEMGVQZklZYWtxYy9hT1ZsCmR6c2cxd3ZDcFdFbWdjUlZHSWJkRFpCb3UwZEhCcmZTdllndWQ3OFc3TDhwUTZvV1lzRkpjKzc5WFJwY0JTRWUKeC9sci81K2RDbmkrR0V6Vm5YcXZrRTZQNUJLWkVqUXNWRWg5RU00eXI3QzlLMHZka1RPS3hUWkJLMlhpeHVKbAoyeEVVRXNEWWNESFk0VFRacmhpMU9aTzdlMEdYV0M2UnM1MThtVjB5bUxhampvcE1oZkpKbzlvdHJCUW1aUkdVCmlrTnpUK042Rk03Z29HWWhZRUdueWVzWGRiRFFPazhDMFBoK3EvNGJYU2NMNENRYTJIeEVhUGkrNnNGZlRXTkgKOG1XY0dMeHdodllJeTF1TnVOTkZlVkwrNjEyS0VmaFkzR0p2Rk01OEJEd3JuNkVnempmUnFGTHE3SzJ0ZHQzOQpmSlBLK1hqUUNVMEtNdXRUdStSV25rd3hPUlFpRldzQ0F3RUFBYU5DTUVBd0RnWURWUjBQQVFIL0JBUURBZ0trCk1BOEdBMVVkRXdFQi93UUZNQU1CQWY4d0hRWURWUjBPQkJZRUZDeEtaZjFjYmcvM3FBZ09IdzFsVEovUCs1RHYKTUEwR0NTcUdTSWIzRFFFQkN3VUFBNElCZ1FBRE9wdytkUGx6d0JjRWU1NDMwRlh3dnlrT1k4YzVoQkRQRGhsUwo4UEVzYU9GVW8xQW9qL2ZuY3VneHFXMUtqMS82MzEycU5qejQ4bVVJTXY3U3h3MFJLQ3hkalZseHRNWk5SM1g2Ci9Sa05SL0R4ZmxOdVFaWGdtNmJwN1hxS3hMKzFvcXltWWU1UFI4MHEwZDV6ZkJOdlQrMUxUaEYwSDRaTE0wQ20KTGhvWEc0LzQ0MDkwbVNOSXJQQStkcXlTNks4YTE3S29DcEJIc3B3V0tFUjRwSms4eDl0MVZaY0pkeEJYb2tKUwpzNzVyTlNCRUpoa0VCZDlKbDZrQi9oc1JvcUdLaTNMcGFONUJQai95VzlxdjAwcHFlRkJrWFdjMUtIK2tyYTRHCnErUi9KQlE0TmRiQ2NJM1pDWjZ5RS9ZY0M5R1dNcGtpMWk1bm94MDNnbVRTem5UdFpOSkdSQXpnbGRVNFI4c2gKOWZDQXV2ejU4RUsxNHZPUzlZRm5XSzVsbGNaZ3lUU29TUGxrWnMzNUd1N2Q5VlN5d1M1NlJiOUdXV2FRcUFSRgpJdkcxbUc0ZnBtMjFFZzJncVhxbHVuUUVpZDk1VFJWdGNLdWc2aWpqTU1UKzJjT2dHVm0ycXcrR282NWZ4OW5jCmpJZExoMWZvZTZ1NUFqMDltS1pyRThMaUlWcz0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
  tls.crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUVYRENDQXNTZ0F3SUJBZ0lSQUpBOXZkRTBnNm1qWTNZQ0gybmxGb2N3RFFZSktvWklodmNOQVFFTEJRQXcKR2pFWU1CWUdBMVVFQXhNUGNtOXZkR05oTFdsdWRHVnlibUZzTUI0WERUSTBNRFV5TVRFek1ESTFNVm9YRFRJMQpNRGd5TURFNU1ESTFNVm93QURDQ0FTSXdEUVlKS29aSWh2Y05BUUVCQlFBRGdnRVBBRENDQVFvQ2dnRUJBT1BrCnlBeGdDVWUwRDV1VE9vK2xUdFhlWjRQZEU3M1o4aGQ0VU9CRUVranZEdUQ3NmpGdkJ6THNYdnZ5V0RmbEJodU0KWkI1bmVQQndMSEsxVmUyekttUkZmL1R4Z1A5b09kdHhkQmdiNlVKc2k3Sk1sN0FSOUdQMUJMblFFcEwwcjA2VApZemJPREJNeGY0T2k1QjhNT2d5V29YNFByc2lBZU53QUp0Wld4SVJlTW9rR1FBbFZyZ3NKU3pPQnF5aDVLc29DCkxha21qWjdYTlJ4STdaMXp0OG9hMlI4bGxEbVkwb1VMUGZDdHlwdW5iNnA2NG9TamNldjhOUTV4Ui9JbjVleWkKbUFzQ292UVlyZ3RnRzFsMTd4YU9KSlQ1VjNQY1ZaUlRncHgzSVM4aU1lRDhLUVR1S3MrRWU1QWN6eEZ1Qko2LwphZ3VyNkxBNE1UZFhUS2ZIb2tNQ0F3RUFBYU9DQVRVd2dnRXhNQTRHQTFVZER3RUIvd1FFQXdJRm9EQVRCZ05WCkhTVUVEREFLQmdnckJnRUZCUWNEQVRBTUJnTlZIUk1CQWY4RUFqQUFNQjhHQTFVZEl3UVlNQmFBRkN4S1pmMWMKYmcvM3FBZ09IdzFsVEovUCs1RHZNSUhhQmdOVkhSRUJBZjhFZ2M4d2djeUNKSE5oYlhCc1pTMXRiMjVwZEc5eQphVzVuTFhOMFlXTnJMV0ZzWlhKMGJXRnVZV2RsY29JdWMyRnRjR3hsTFcxdmJtbDBiM0pwYm1jdGMzUmhZMnN0CllXeGxjblJ0WVc1aFoyVnlMbTl3WlhKaGRHOXljNEl5YzJGdGNHeGxMVzF2Ym1sMGIzSnBibWN0YzNSaFkyc3QKWVd4bGNuUnRZVzVoWjJWeUxtOXdaWEpoZEc5eWN5NXpkbU9DUUhOaGJYQnNaUzF0YjI1cGRHOXlhVzVuTFhOMApZV05yTFdGc1pYSjBiV0Z1WVdkbGNpNXZjR1Z5WVhSdmNuTXVjM1pqTG1Oc2RYTjBaWEl1Ykc5allXd3dEUVlKCktvWklodmNOQVFFTEJRQURnZ0dCQUpPbDhjWHgzb2NZVWRRY2tNNlo0L3NlMXdCOWczaTZ6cUJmZXk0ckN0VzYKQThjdEFpNkRHM1o3MC9LRnlnVGNSRWYwSWRGQStXYzB5NkNkOVV6R1piY05sd3ZFdDNjV01yYUx0ZmlTL2w1LwpmaXdXWHgyei91N1VYL0tDM2k2UXIwQVRIanZGTHVlSFVJbGlZMGtoT1JGc3lJbkl4LzlOZUV4d0IxRXBIUHAzCmpCUWRRamMzaHU5Z1FkRnc4dUxqUUh3Ujd0TFQ1c20yOS8zRU1aQWtaQndYakNHcXVWYzRDaDdONmdiUUFLYTgKVHhqUzJGWDBMNWMxbXA2S01Gbk9oeVVuNGtBUWk3R1hHbGdhcUV3TDVUTGZOQXQzTkpQc0Nzcm8rLzFnSHhVRAo0aE44WE1vNXIvRjduRm9DNFRXSEoza1VSVG1BNkM0dDRBMXd3cTB4YTllS29BcjZNT1RDaHFSN1pWY1BkczZxCnY3SlFtRVdqR0pKL1VIQktxalk1NTQxdzMzdlVCSDZKdm1jYmRGNzdvSzRTUkdVQ2xDWk5kQU4rTkZLODVtcXQKYWs3dUxhR05RVFpReGJ0UzFpdVI3TjBORjg3UDUvemFEcHFUVTNnWnp3cTNzbVVmTTRZWnh4QVJUZ2ZFY2RWQQppMTNDczhWMng1aWRNeGR0eXNvMVp3PT0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
  tls.key: LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQpNSUlFcEFJQkFBS0NBUUVBNCtUSURHQUpSN1FQbTVNNmo2Vk8xZDVuZzkwVHZkbnlGM2hRNEVRU1NPOE80UHZxCk1XOEhNdXhlKy9KWU4rVUdHNHhrSG1kNDhIQXNjclZWN2JNcVpFVi85UEdBLzJnNTIzRjBHQnZwUW15THNreVgKc0JIMFkvVUV1ZEFTa3ZTdlRwTmpOczRNRXpGL2c2TGtId3c2REphaGZnK3V5SUI0M0FBbTFsYkVoRjR5aVFaQQpDVld1Q3dsTE00R3JLSGtxeWdJdHFTYU5udGMxSEVqdG5YTzN5aHJaSHlXVU9aalNoUXM5OEszS202ZHZxbnJpCmhLTng2L3cxRG5GSDhpZmw3S0tZQ3dLaTlCaXVDMkFiV1hYdkZvNGtsUGxYYzl4VmxGT0NuSGNoTHlJeDRQd3AKQk80cXo0UjdrQnpQRVc0RW5yOXFDNnZvc0RneE4xZE1wOGVpUXdJREFRQUJBb0lCQUVEaUt6Nnl2UkdvSjJWdwo5cFJkU1dQRXpZNFNqSU5hSVA0anFFc2h0bDhEaVNYdk0wOWR3QnYwaGxxUE5pUzcxUkJVMlZCNnYwa2RkdG8xCjNNQXZhWnRnMmp4LzRlMVYzRnNwSXhlV0lzUmhxU2hiZkRpSjlCM2RiYm9GanJRK2JBY2dHWGhLRTlBSElMbjIKejZJelZmZXRqMDhYNGZoOTgyc1hzL2NFZnpEOEl5c2NRazdzR2xMTEc2ODhRbXhXWnBZZlhhR3o0NUF1RGZMSQpNek16VzVlWEVwM1hiWWVlUGlYVWQydDE3SThWRWtWZkZuR0NGTG9Gb3U2MUZEQjZRSmE3VEdCVnpKUGh2c1ZICjZKRjlvWjhTVWZXNTl2RTd1cjhMUHR2MElQV3BBcTlDS21aS3RLS2pFdW1rSUpYeHNVSzZ1amxLN2RhRFZzaFMKZjFkZk5ya0NnWUVBNjBIWFlYVStZcXJEZytJMXhRUUgzTWgreFpsV2lkVFVwVFgrWUVueE1UTGM3Nk1CQUlzZQpUN0xtMGNobmx0YmllYy9HZVhlUmZLT2VLZXlsWVBON3ZYU05Nelp0ZFNnZXJOSU90dDFrSzRMMG5XKytxS3U4Cm44OFMrME1ueWlTeVpMTHgzd2pRR0lOd0FEVEpsVk1PU29yak9BY0pQV0ZzUDIvNkk3L09kQzhDZ1lFQTkveTcKb1UyR0Y5aHdmdkMxZ1lNajBNM3cxd2NOVnRXMzVZT2RrbENVY21IT3JvdWF5Rm9KR3ZsS1hrMGJoTytERW96aApZd1VUOVVEN0pxSy85NmdwanFlL2hQZXRCbnJZcElnaEswZmp5WGRyZUQ2MVNkcnp2T3Ywd1VJSEFvQmpKYURrClJwcDhMV3F4Y2FXZU1DY0drbTlZUUdWMGJtWWxRYy9NUGFxS3FpMENnWUVBNFA5MGxWSURtMndFVitNR1YrbjEKOXhPT0l6LzNBeFdaVm5UZ1Y0a0FjM1JZMTV2SFBwVFBKWnhObzMrT2l6WHpIbHM0VUQycDJGSEgyQUxyWEdFaQpERjBuR2o3SXlzN1hINlE2ekd2Sy9NNWZVeDluY252bGg5WU5LbC9jOHpqTG9RRi8wRGJtaHp0VjUreVVpd3NIClZRV21NSlNTSXJGRlRScTBXbko4YXFNQ2dZRUFpZWJGQUNINEczS0V2RkptM0twc1ZFVHhWT0hiRFJxenRMSEkKQWdGRlM0WVFSUWZrL3R6eWkwbHRqWnNiWG9WQ1FpS05nMTdqaGxLb1Z3ZlI4SXBXOGJsZXl6WmZ4V2F5Nm11NAo1alNIK3hTcGNxaU1iNUdPVU5SMXVySlk2NDh4azA4eElnSHVZK01jNGJIVGtidGg4RFJ5d3Z5b0RteEc4NTMyCldhUm9qOUVDZ1lCakl4ZEpvRGs1UnliaUZBbWhNS0tpN0tVR3pBeXZtaW5pSmdFVE95dTJ6Umo0cWwwaDNRQjIKNEZXbEUydHhmSEtpY0Y1eGhVRVBDRlQ2T0pHLzdLVWFQbks3RkVYTFVBWGRzSGhwZkVYdWFlTko2TnI1bVh4VwppcVViYjBTOWhXNGxUaWQ5VS9jOTRmWGZXYkhLb2ljcE10S0pZRVNYTHZIWVE4VVFSa3dSYWc9PQotLS0tLUVORCBSU0EgUFJJVkFURSBLRVktLS0tLQo=
kind: Secret
metadata:
  annotations:
    cert-manager.io/alt-names: sample-monitoring-stack-alertmanager,sample-monitoring-stack-alertmanager.operators,sample-monitoring-stack-alertmanager.operators.svc,sample-monitoring-stack-alertmanager.operators.svc.cluster.local
    cert-manager.io/certificate-name: alertmanager-svc
    cert-manager.io/common-name: ""
    cert-manager.io/ip-sans: ""
    cert-manager.io/issuer-group: cert-manager.io
    cert-manager.io/issuer-kind: Issuer
    cert-manager.io/issuer-name: rootca-internal
    cert-manager.io/uri-sans: ""
  creationTimestamp: "2024-05-21T13:02:51Z"
  labels:
    controller.cert-manager.io/fao: "true"
  name: cert-alertmanager-svc
  namespace: operators
  resourceVersion: "410607"
  uid: 46857266-f8ca-42a4-a655-2dbfe0359687
type: kubernetes.io/tls
```

### prometheus-operator log output

```shell
kubectl logs prometheus-sample-monitoring-stack-0 -p
ts=2024-07-18T13:23:12.382Z caller=main.go:633 level=info msg="Starting Prometheus Server" mode=server version="(version=2.53.0, branch=HEAD, revision=4c35b9250afefede41c5f5acd76191f90f625898)"
ts=2024-07-18T13:23:12.382Z caller=main.go:638 level=info build_context="(go=go1.22.4, platform=linux/amd64, user=root@7f8d89cbbd64, date=20240619-07:39:12, tags=netgo,builtinassets,stringlabels)"
ts=2024-07-18T13:23:12.382Z caller=main.go:639 level=info host_details="(Linux 5.14.0-386.el9.x86_64 #1 SMP PREEMPT_DYNAMIC Wed Nov 15 10:38:22 UTC 2023 x86_64 prometheus-sample-monitoring-stack-0 (none))"
ts=2024-07-18T13:23:12.382Z caller=main.go:640 level=info fd_limits="(soft=524288, hard=524288)"
ts=2024-07-18T13:23:12.382Z caller=main.go:641 level=info vm_limits="(soft=unlimited, hard=unlimited)"
ts=2024-07-18T13:23:12.387Z caller=web.go:568 level=info component=web msg="Start listening for connections" address=0.0.0.0:9090
ts=2024-07-18T13:23:12.387Z caller=main.go:1148 level=info msg="Starting TSDB ..."
ts=2024-07-18T13:23:12.389Z caller=tls_config.go:313 level=info component=web msg="Listening on" address=[::]:9090
ts=2024-07-18T13:23:12.389Z caller=tls_config.go:352 level=info component=web msg="TLS is disabled." http2=false address=[::]:9090
ts=2024-07-18T13:23:12.395Z caller=head.go:626 level=info component=tsdb msg="Replaying on-disk memory mappable chunks if any"
ts=2024-07-18T13:23:12.395Z caller=head.go:713 level=info component=tsdb msg="On-disk memory mappable chunks replay completed" duration=2.377µs
ts=2024-07-18T13:23:12.395Z caller=head.go:721 level=info component=tsdb msg="Replaying WAL, this may take a while"
ts=2024-07-18T13:23:12.399Z caller=head.go:793 level=info component=tsdb msg="WAL segment loaded" segment=0 maxSegment=9
ts=2024-07-18T13:23:12.402Z caller=head.go:793 level=info component=tsdb msg="WAL segment loaded" segment=1 maxSegment=9
ts=2024-07-18T13:23:12.405Z caller=head.go:793 level=info component=tsdb msg="WAL segment loaded" segment=2 maxSegment=9
ts=2024-07-18T13:23:12.407Z caller=head.go:793 level=info component=tsdb msg="WAL segment loaded" segment=3 maxSegment=9
ts=2024-07-18T13:23:12.408Z caller=head.go:793 level=info component=tsdb msg="WAL segment loaded" segment=4 maxSegment=9
ts=2024-07-18T13:23:12.409Z caller=head.go:793 level=info component=tsdb msg="WAL segment loaded" segment=5 maxSegment=9
ts=2024-07-18T13:23:12.412Z caller=head.go:793 level=info component=tsdb msg="WAL segment loaded" segment=6 maxSegment=9
ts=2024-07-18T13:23:12.413Z caller=head.go:793 level=info component=tsdb msg="WAL segment loaded" segment=7 maxSegment=9
ts=2024-07-18T13:23:12.414Z caller=head.go:793 level=info component=tsdb msg="WAL segment loaded" segment=8 maxSegment=9
ts=2024-07-18T13:23:12.415Z caller=head.go:793 level=info component=tsdb msg="WAL segment loaded" segment=9 maxSegment=9
ts=2024-07-18T13:23:12.415Z caller=head.go:830 level=info component=tsdb msg="WAL replay completed" checkpoint_replay_duration=96.223µs wal_replay_duration=20.315289ms wbl_replay_duration=183ns chunk_snapshot_load_duration=0s mmap_chunk_replay_duration=2.377µs total_replay_duration=20.480246ms
ts=2024-07-18T13:23:12.418Z caller=main.go:1169 level=info fs_type=XFS_SUPER_MAGIC
ts=2024-07-18T13:23:12.418Z caller=main.go:1172 level=info msg="TSDB started"
ts=2024-07-18T13:23:12.418Z caller=main.go:1173 level=debug msg="TSDB options" MinBlockDuration=2h MaxBlockDuration=2h24m MaxBytes=0B NoLockfile=false RetentionDuration=1d WALSegmentSize=0B WALCompression=true
ts=2024-07-18T13:23:12.418Z caller=main.go:1354 level=info msg="Loading configuration file" filename=/etc/prometheus/config_out/prometheus.env.yaml
ts=2024-07-18T13:23:12.419Z caller=main.go:1380 level=error msg="Failed to apply configuration" err="unable to read CA cert: unable to read file /etc/prometheus/certs/0__cert-alertmanager-svc_ca.crt: open /etc/prometheus/certs/0__cert-alertmanager-svc_ca.crt: no such file or directory"
ts=2024-07-18T13:23:12.420Z caller=kubernetes.go:331 level=info component="discovery manager notify" discovery=kubernetes config=config-0 msg="Using pod service account via in-cluster config"
ts=2024-07-18T13:23:12.420Z caller=manager.go:170 level=debug component="discovery manager notify" msg="Starting provider" provider=kubernetes/0 subs=[config-0]
ts=2024-07-18T13:23:12.420Z caller=main.go:1007 level=info msg="Stopping scrape discovery manager..."
ts=2024-07-18T13:23:12.420Z caller=main.go:1021 level=info msg="Stopping notify discovery manager..."
ts=2024-07-18T13:23:12.420Z caller=manager.go:178 level=info component="rule manager" msg="Stopping rule manager..."
ts=2024-07-18T13:23:12.420Z caller=manager.go:188 level=info component="rule manager" msg="Rule manager stopped"
ts=2024-07-18T13:23:12.420Z caller=main.go:1058 level=info msg="Stopping scrape manager..."
ts=2024-07-18T13:23:12.420Z caller=manager.go:164 level=info component="rule manager" msg="Starting rule manager..."
ts=2024-07-18T13:23:12.420Z caller=main.go:1003 level=info msg="Scrape discovery manager stopped"
ts=2024-07-18T13:23:12.420Z caller=main.go:1050 level=info msg="Scrape manager stopped"
ts=2024-07-18T13:23:12.420Z caller=main.go:1017 level=info msg="Notify discovery manager stopped"
ts=2024-07-18T13:23:12.420Z caller=klog.go:55 level=debug component=k8s_client_runtime func=Verbose.Infof msg="Starting reflector *v1.Service (0s) from pkg/mod/k8s.io/client-go@v0.29.3/tools/cache/reflector.go:229"
ts=2024-07-18T13:23:12.420Z caller=klog.go:55 level=debug component=k8s_client_runtime func=Verbose.Infof msg="Starting reflector *v1.Pod (0s) from pkg/mod/k8s.io/client-go@v0.29.3/tools/cache/reflector.go:229"
ts=2024-07-18T13:23:12.420Z caller=klog.go:55 level=debug component=k8s_client_runtime func=Verbose.Infof msg="Stopping reflector *v1.Service (0s) from pkg/mod/k8s.io/client-go@v0.29.3/tools/cache/reflector.go:229"
ts=2024-07-18T13:23:12.420Z caller=klog.go:55 level=debug component=k8s_client_runtime func=Verbose.Infof msg="Stopping reflector *v1.Pod (0s) from pkg/mod/k8s.io/client-go@v0.29.3/tools/cache/reflector.go:229"
ts=2024-07-18T13:23:12.420Z caller=klog.go:55 level=debug component=k8s_client_runtime func=Verbose.Infof msg="Starting reflector *v1.Endpoints (0s) from pkg/mod/k8s.io/client-go@v0.29.3/tools/cache/reflector.go:229"
ts=2024-07-18T13:23:12.420Z caller=klog.go:55 level=debug component=k8s_client_runtime func=Verbose.Infof msg="Stopping reflector *v1.Endpoints (0s) from pkg/mod/k8s.io/client-go@v0.29.3/tools/cache/reflector.go:229"
ts=2024-07-18T13:23:12.421Z caller=notifier.go:623 level=info component=notifier msg="Stopping notification manager..."
ts=2024-07-18T13:23:12.421Z caller=main.go:1280 level=info msg="Notifier manager stopped"
ts=2024-07-18T13:23:12.421Z caller=main.go:1289 level=error err="error loading config from \"/etc/prometheus/config_out/prometheus.env.yaml\": one or more errors occurred while applying the new configuration (--config.file=\"/etc/prometheus/config_out/prometheus.env.yaml\")"
```

### Relevant commands from debug pod by executing `kubectl debug prometheus-sample-monitoring-stack-0 --container prometheus --copy-to debugpod -ti -- /bin/sh`

```
# Notice the ca_file path
$ cat /etc/prometheus/config_out/prometheus.env.yaml 
global:
  evaluation_interval: 30s
  scrape_interval: 30s
  external_labels:
    prometheus: operators/sample-monitoring-stack
    prometheus_replica: debugpod
rule_files:
- /etc/prometheus/rules/prometheus-sample-monitoring-stack-rulefiles-0/*.yaml
scrape_configs: []
alerting:
  alert_relabel_configs:
  - action: labeldrop
    regex: prometheus_replica
  alertmanagers:
  - path_prefix: /
    scheme: https
    tls_config:
      ca_file: /etc/prometheus/certs/0__cert-alertmanager-svc_ca.crt
      server_name: sample-monitoring-stack-alertmanager
    kubernetes_sd_configs:
    - role: endpoints
      namespaces:
        names:
        - operators
    api_version: v2
    relabel_configs:
    - action: keep
      source_labels:
      - __meta_kubernetes_service_name
      regex: sample-monitoring-stack-alertmanager
    - action: keep
      source_labels:
      - __meta_kubernetes_endpoint_port_name
      regex: web
      
$ ls -la /etc/prometheus/certs
total 0
drwxrwxrwt    3 root     root            80 Jul 18 13:33 .
drwxr-xr-x    1 nobody   nobody          88 Jul 18 13:34 ..
drwxr-xr-x    2 root     root            40 Jul 18 13:33 ..2024_07_18_13_33_47.1471525806
lrwxrwxrwx    1 root     root            32 Jul 18 13:33 ..data -> ..2024_07_18_13_33_47.1471525806
```

### Related volumes and volumemounts from the Prometheus pod
```
kubectl get pod prometheus-sample-monitoring-stack-0 -ojsonpath --template={.spec.containers[0].volumeMounts} | jq
[
...
  {
    "mountPath": "/etc/prometheus/certs",
    "name": "tls-assets",
    "readOnly": true
  },
...
]

$ kubectl get pod prometheus-sample-monitoring-stack-0 -ojsonpath --template={.spec.volumes} | jq
[
...
  {
    "name": "tls-assets",
    "projected": {
      "defaultMode": 420,
      "sources": [
        {
          "secret": {
            "name": "prometheus-sample-monitoring-stack-tls-assets-0"
          }
        }
      ]
    }
  },
...
]

```

### The empty tls-assets secret
```
kubectl describe secret/prometheus-sample-monitoring-stack-tls-assets-0
Name:         prometheus-sample-monitoring-stack-tls-assets-0
Namespace:    operators
Labels:       managed-by=prometheus-operator
Annotations:  <none>

Type:  Opaque

Data
====
```

**Comment by simonpasquier** [comment]
Thanks for the report! The bug has been introduced by 16d2298f1937d1c039075ae0cf00d0421da604d8. I'll push a fix to the release-0.75 branch.

**Comment by mviswanathsai** [comment]
Has a fix been pushed?

**Comment by simonpasquier** [comment]
The fix is #6765 but I referenced a wrong issue number.
